kernel/dev/serial/rs232.cpp

```diff
@@ -101,3 +101,44 @@ px_ring_buff_t* px_rs232_init_buffer(int size) {
         return NULL;
     }
 }
+
+px_ring_buff_t* px_rs232_get_buffer() {
+    // Can return NULL. This is documented.
+    return read_buffer;
+}
+
+char px_rs232_get_char() {
+    // Grab the last byte and convert to a char
+    uint8_t data;
+    px_ring_buffer_dequeue(read_buffer, &data);
+    return (char)data;
+}
+
+char* px_rs232_get_str() {
+    int idx = 0;
+    char* str = (char*)malloc(sizeof(char) * 1024);
+    // Keep reading until the buffer is empty or
+    // a newline is read.
+    while (!px_ring_buffer_is_empty(read_buffer)) {
+        uint8_t byte;
+        px_ring_buffer_dequeue(read_buffer, &byte);
+        str[idx] = (char)byte;
+        ++idx;
+        // Break if it's a newline
+        if ((char)byte == '\n') {
+            break;
+        }
+    }
+    // Add the null terminator
+    str[idx] = '\0';
+    // Return the string
+    return str;
+}
+
+int px_rs232_close() {
+    int ret = -1;
+    if (read_buffer != NULL) {
+        ret = px_ring_buffer_destroy(read_buffer);
```

Comment:
Should we also set `read_buffer` to `null` since it no longer is useful?

Reply:
It gets freed in the destroy call. We can just set it to NULL

Reply:
Resolved in the latest commit (hopefully)